src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueHealthFactorLogic.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+// import { Math } from "src/utils/Math.sol";
+import { IMorpho, MarketParams, Market, Id } from "src/interfaces/external/Morpho/MorphoBlue/interfaces/IMorpho.sol";
+import { MathLib, WAD } from "src/interfaces/external/Morpho/MorphoBlue/libraries/MathLib.sol";
+import { SharesMathLib } from "src/interfaces/external/Morpho/MorphoBlue/libraries/SharesMathLib.sol";
+import { IOracle } from "src/interfaces/external/Morpho/MorphoBlue/interfaces/IOracle.sol";
+import { UtilsLib } from "src/interfaces/external/Morpho/MorphoBlue/libraries/UtilsLib.sol";
+import { MorphoLib } from "src/interfaces/external/Morpho/MorphoBlue/libraries/periphery/MorphoLib.sol";
+import { console } from "@forge-std/Test.sol";
+
+/**
+ * @title Morpho Blue Health Factor Logic contract.
+ * @notice Implements health factor logic used by both
+ *         the MorphoBlueCollateralAdaptor && MorphoBlueDebtAdaptor.
+ * @author crispymangoes, 0xEinCodes
+ * NOTE: helper functions made virtual in case future Morpho Blue Pair versions require different implementation logic.
+ * NOTE: The library from Morpho provides exposed getters give cellar totalSupply and totalBorrow with interest accrued, although they are simulated values they have been tested compared to their actual Morpho Blue prod contract (within their test suite). This helper contract DOES NOT USE THE LIBRARIES FOR NOW and will try them during the testing phase of development. We will see if the libraries are more gas-efficient vs using getters. QUESTION FOR MORPHO TEAM - library for balances is usable and accurate to depend on. Do we want any failsafes just in case?
+ */
+contract MorphoBlueHealthFactorLogic {
+    // using Math for uint256;
+
+    // libraries from Morpho Blue codebase to ensure same mathematic methods for HF calcs
+    using MathLib for uint128;
+    using MathLib for uint256;
+    using UtilsLib for uint256;
+    using SharesMathLib for uint256;
+    using MorphoLib for IMorpho;
+
+    /**
+     * @notice The Morpho Blue contract on current network.
+     */
+    IMorpho public immutable morphoBlue;
+
+    // Constant from MorphoBlue
+    uint256 constant ORACLE_PRICE_SCALE = 1e36;
+
+    constructor(address _morphoBlue) {
+        morphoBlue = IMorpho(_morphoBlue);
+    }
+
+    /**
+     * @notice The ```_getHealthFactor``` function returns the current health factor of a respective position given an exchange rate
+     * @param _id The specified Morpho Blue market Id
+     * @param _market The specified Morpho Blue market
+     * @return currentHF The health factor of the position atm
+     */
+    function _getHealthFactor(Id _id, MarketParams memory _market) internal view virtual returns (uint256 currentHF) {
```

Comment:
This function does not kick the market because the interest should already be accrued from some strategist action before right? Like taking on a new borrow, or repaying debt, or removing collateral?

Reply:
Correct, this should not kick the market as other functions that subsequently kick the market happen beforehand (borrow, repay, remove collateral). The mentioned actions would be the ones calling the health factor check, whereas things like adding collateral, or lending, or withdrawing loan tokens would not call this.